console/src/components/Main/Main.js

```diff
@@ -35,10 +35,45 @@ import {
   setProClickState,
   getLoveConsentState,
   setLoveConsentState,
+  getUserType,
 } from './utils';
 import { getSchemaBaseRoute } from '../Common/utils/routesUtils';
 import LoveSection from './LoveSection';
 import { Help, ProPopup } from './components/';
+import { UPDATE_DATA_HEADERS } from '../Services/Data/DataActions';
+import { HASURA_COLLABORATOR_TOKEN } from '../../constants';
+import { UPDATE_CONSOLE_NOTIFICATIONS } from '../../telemetry/Actions';
+
+const updateRequestHeaders = props => {
```

Comment:
Should notification be refreshed whenever any header is changed in the API explorer tab? The state at `state.apiexplorer.displayedApi.request.headers` would be updated on any change in the `GraphiQL` header tab. The notification section should be dependent on the headers (stored in `~table.dataHeaders`) IMO.

Reply:
Yeah, fair point. Since these changes are ending up on Cloud and/or Pro and since `state.table.dataHeaders` has the collaborator token already. I can make this change.

Reply:
@karthikvt26, I have pushed the changes related to this. You can take another look.